src/main/java/services/gorest/stepdefinition/PostsStepsDefinition.java

```diff
@@ -0,0 +1,99 @@
+package services.gorest.stepdefinition;
+
+
+import cucumber.api.Scenario;
+import cucumber.api.java.After;
+import cucumber.api.java.Before;
+import cucumber.api.java.en.When;
+import io.restassured.response.Response;
+import net.thucydides.core.annotations.Steps;
+import services.gorest.actions.post.CreatePost;
+import services.gorest.actions.post.DeletePost;
+import services.gorest.actions.post.GetPost;
+import services.gorest.actions.post.UpdatePost;
+import services.gorest.actions.user.CreateUser;
+import services.gorest.actions.user.DeleteUser;
+import services.gorest.models.responses.GetPostResponse;
+import services.gorest.models.responses.GetUserResponse;
+import utils.methods.ReusableMethods;
+
+import static utils.methods.ReusableMethods.replaceExpectedWithVariable;
+import static utils.variables.SessionVariableManager.getSessionVariable;
+import static utils.variables.SessionVariableManager.setSessionVariable;
+import static utils.variables.SessionVariables.*;
+
+public class PostsStepsDefinition {
+
+    @Steps
+    private CreatePost createPost;
+
+    @Steps
+    private GetPost getPost;
+
+    @Steps
+    private UpdatePost updatePost;
+
+    @Steps
+    private DeletePost deletePost;
+
+    @Steps
+    private ReusableMethods reusableMethods;
+
+    @Steps
+    private CreateUser createUser;
+
+    @Steps
+    private DeleteUser deleteUser;
+
+    @Before("@PostSmoke")
+    public void createPrereq() {
+        Response response = createUser.whenCreateRandomUserObject();
+        setSessionVariable(VAR_USER_ID, response.as(GetUserResponse.class).getResult().getId());
+    }
+
+    @When("^I create a new post with my user id (.*), I provide the title (.*) and add the following body:$")
+    public void whenCreatePost(String userId, String title, String body) {
+        int uId = Integer.valueOf(replaceExpectedWithVariable(userId, VAR_USER_ID));
+        Response response = createPost.whenCreateNewPost(uId, title, body);
+        GetPostResponse getPostResponse = response.as(GetPostResponse.class);
+        setSessionVariable(VAR_RESPONSE, response);
+        if (getSessionVariable(VAR_RESPONSE) != null) {
```

Comment:
you just set above the response. why do here the check? we check a value is != null when we need to use it there. lets discuss this offline